plugins/services/src/js/components/ServiceStatusWarning.js

```diff
@@ -7,27 +7,42 @@ import Icon from '../../../../../src/js/components/Icon';
 import Pod from '../structs/Pod';
 import Service from '../structs/Service';
 import ServiceTree from '../structs/ServiceTree';
+import StringUtil from '../../../../../src/js/utils/StringUtil';
 
 class ServiceStatusWarning extends Component {
-  getTooltipContent(timeWaiting) {
-    return `DC/OS has been waiting for resources and is unable to complete this deployment for ${DateUtil.getDuration(timeWaiting, null)}.`;
+  getTooltip(content) {
+    return (
+      <Tooltip
+        content={content}
+        width={250}
+        wrapText={true}
+        wrapperClassName="tooltip-wrapper status-waiting-indicator">
+        <Icon color="red" id="yield" size="mini" />
+      </Tooltip>
+    );
   }
+
   render() {
     const {item} = this.props;
-    const queue = item.getQueue();
-
-    if (DeclinedOffersUtil.shouldDisplayDeclinedOffersWarning(queue)) {
-      const timeWaiting = DeclinedOffersUtil.getTimeWaiting(queue);
-
-      return (
-        <Tooltip
-          content={this.getTooltipContent(timeWaiting)}
-          width={250}
-          wrapText={true}
-          wrapperClassName="tooltip-wrapper status-waiting-indicator">
-          <Icon color="red" id="yield" size="mini" />
-        </Tooltip>
-      );
+
+    if (item instanceof ServiceTree) {
+      const appsWithDeclinedOffers = DeclinedOffersUtil
```

Comment:
There is no need of a new util if you only want to reduce the service tree to a list of apps with declined offers...

*List*
```
    const appsWithDeclinedOffers = item.filterItems(function (item) {
        if (!(item instanceof ServiceTree)) {
          return DeclinedOffersUtil
            .shouldDisplayDeclinedOffersWarning(item.getQueue());
        }

        return false;
      }).flattenItems();
```